src/Widgets/DataWidget/DataWidgetComponent.tsx

```diff
@@ -7,14 +7,36 @@ provideComponent(
   DataWidget,
   ({ widget }) => {
     const dataScope = useData()
+    let dataError: unknown
 
-    if (dataScope.isEmpty()) {
-      return <EditorNote>Data is empty.</EditorNote>
+    try {
+      if (dataScope.isEmpty()) return <EditorNote>Data is empty.</EditorNote>
+    } catch (error) {
+      dataError = error
+    }
+
+    let dataItems: DataItem[]
+    try {
+      dataItems = dataScope.take()
+    } catch (error) {
+      dataItems = []
+      dataError = error
+    }
+
+    if (dataError) {
+      return (
+        <EditorNote>
+          Error fetching data:{' '}
+          {dataError instanceof Error
+            ? dataError.message
+            : JSON.stringify(dataError)}
+        </EditorNote>
```

Comment:
* [ ] FYI this code is duplicated quite a bit. How about extracting a dedicated component?